cargo/cargo_build_script_runner/lib.rs

```diff
@@ -230,24 +235,27 @@ mod tests {
     fn test_from_read_buffer_to_env_and_flags() {
         let buff = Cursor::new(
             "
-cargo:rustc-link-lib=sdfsdf
-cargo:rustc-env=FOO=BAR
-cargo:rustc-link-search=/some/absolute/path/bleh
-cargo:rustc-env=BAR=FOO
-cargo:rustc-flags=-Lblah
-cargo:rerun-if-changed=ignored
-cargo:rustc-cfg=feature=awesome
-cargo:version=123
-cargo:version_number=1010107f
-cargo:include_path=/some/absolute/path/include
-cargo:rustc-env=SOME_PATH=/some/absolute/path/beep
-cargo:rustc-link-arg=-weak_framework
-cargo:rustc-link-arg=Metal
-cargo:rustc-env=no_trailing_newline=true",
+cargo::rustc-link-lib=sdfsdf
```

Comment:
Can you actually keep some test data where the single `:` case is checked? Lots of crates still use that so it's important that it still works (at least until Cargo decides to fully deprecate it).

Reply:
Yes, a test for the single-colon case was already included in my commit; check the bottom of the paragraph for the `cargo:rustc-env=old_cargo_colon_prefix_works=true` on line 254.

I also added two new tests for non-cargo-instruction lines; I saw the logic checking for them, but there was no test I found covering it, so I added some.

Reply:
Ah, I missed that. Sweet!